### PR TITLE
Allow ‘=‘ character in CLI inputs

### LINKED ISSUE #5131: CLI inputs do not handle = (equals) character
Present from 4.18 onwards. 

To reproduce:

```
control 'Check an inputs value' do
  describe input(:my_input) do
    it { should eq 'ab=cde' }
  end
end
```

```
jstocks@phobos profiles % inspec exec my_profile --input my_input='ab=cde'

Profile: InSpec Profile (my_profile)
Version: 0.1.0
Target:  local://

  ×  Check an inputs value: ab
     ×  ab is expected to eq "ab=cde"
     
     expected: "ab=cde"
          got: "ab"
     
     (compared using ==)



Profile Summary: 0 successful controls, 1 control failure, 0 controls skipped
Test Summary: 0 successful, 1 failure, 0 skipped

```

It seems to terminate the string on the `=` character.

In 4.20 onwards since we have parsing of inputs to hashes (I thought this might offer a quick workaround), this also does not work - InSpec still treats the YAML/JSON expression as a `=`-terminated string.

Aha! Link: https://chef.aha.io/features/SH-2510

### PR DESCRIPTION
**Pull request by Schwad**
Fixes #5131

Due to the use of `#split(‘=‘)` against inputs supplied via the CLI we had an edge case where inputs with `’=‘` in the value would cause a breakage.

This PR supplies a test for the regression and fixes the bug with a regex solution.

Signed-off-by: Nick Schwaderer <nschwaderer@chef.io>

Aha! Link: https://chef.aha.io/features/SH-2552

### CODE REVIEW

**Review by clintoncwolfe — APPROVED** [review]
Thanks! I like where you put the test fixture profile.

**Comment by Schwad** [comment]
> Thanks! I like where you put the test fixture profile.

@clintoncwolfe Cheers! 😄  I figured that might be a useful pattern for future PRs like this. (Plus, when you're scanning fixtures in future hopefully it makes things clear what the genesis of that profile was)

File: test/fixtures/profiles/regressions/5131_input_equals/controls/input_equals.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+title 'Regressions for 5131'
```
Comment (james-stocks) [inline test/fixtures/profiles/regressions/5131_input_equals/controls/input_equals.rb]: It's only a regression if it previously worked and we broke it. Did this case previously work OK in the past?

Comment (james-stocks) [reply test/fixtures/profiles/regressions/5131_input_equals/controls/input_equals.rb]: When I said `Present from 4.18 onwards.` I meant it's known to be a problem that far back. I didn't mean that it worked previous to that version, sorry.

Comment (Schwad) [reply test/fixtures/profiles/regressions/5131_input_equals/controls/input_equals.rb]: Good point! I've reverted and tried to put this fixture in a more aptly named location. If the pipeline passes I'll give a rebase squash so we're clean again 👍

Comment (james-stocks) [reply test/fixtures/profiles/regressions/5131_input_equals/controls/input_equals.rb]: Thanks. It would be fantastic to have a `regressions` directory for cases that are regressions. Sorry if my issue description was misleading

**Review by james-stocks — APPROVED** [review]
Thank you @Schwad :)
